components/server/src/auth/login-completion-handler.ts

```diff
@@ -93,6 +97,26 @@ export class LoginCompletionHandler {
             );
         }
 
+        const isJWTCookieExperimentEnabled = await getExperimentsClientForBackend().getValueAsync(
+            "jwtSessionCookieEnabled",
+            false,
+            {
+                user: user,
+            },
+        );
+        if (isJWTCookieExperimentEnabled) {
+            const token = await this.authJWT.sign(user.id, {});
+
+            response.cookie(SessionHandlerProvider.getJWTCookieName(this.config.hostUrl), token, {
+                maxAge: this.config.session.maxAgeMs,
+                httpOnly: true,
+                sameSite: "lax",
```

Comment:
Let's use `strict` cookies, please.

Reply:
A strict cookie means, that if the browser (already received the cookie) is redirected to a 3rd party, and then back to Gitpod, the cookie will _not_ be sent. This is a typical scenario for redirect-based flows, where the 3rd party knows some sort of "returnTo" URL to redirect back, e.g. after asking user for consent. 

Hmm, now having that said, it sounds like we'd break Git Integrations with that. There are (at least) two solutions available:
1. using a state param _and_ a nonce cookie for redirect-based flows, similar to what's now done for OIDC SSO. There we no longer rely on a session.
2. use a trick to relax the cookie for the time of a web-flow cycle: https://github.com/gitpod-io/gitpod/pull/16406

That's getting to complicating here, I guess. 

So, not raising the bar here seems ok, but keep in mind that we need to make it strict eventually to improve the security posture.

Reply:
Thanks!